avocado/utils/nvme.py

```diff
@@ -90,3 +93,220 @@ def get_controller_id(controll_name):
+        delete_ns(controller_name, ns_id)
+
+
+def is_ns_exists(controller_name, ns_id):
+    """
+    Returns if that particular namespace exists on the controller or not
+
+    :param controller_name: name of the controller on which we want to check
+                            ns existance
+
+    :returns: True if ests else False
+    :rtype: boolean
+    """
+    ns_list = get_current_ns_list(controller_name)
+    if ns_id in ns_list:
+        return True
+    return False
+
+
+def get_lba(namespace):
+    """
+    Returns LBA of the namespace. If not found, return defaults to 0.
+
+    :param namespace: nvme namespace like /dev/nvme0n1, /dev/nvme0n2 etc..
+    :rtype: Integer
+    """
+    if namespace:
+        cmd = f"nvme id-ns /dev/{namespace}"
+        out = process.run(cmd, shell=True, ignore_status=True)
+        for line in out.splitlines():
+            if "in use" in line:
+                return int(line.split()[1])
+    return "0"
+
+
+def ns_rescan(controller_name):
+    """
+    re-scans all the names spaces on the given contrller
+
+    :param controller_name: controller name on which re-scan is applied
+    """
+    cmd = f"nvme ns-rescan {controller_name}"
+    try:
+        process.run(cmd, shell=True, ignore_status=True)
+    except process.CmdError as detail:
+        LOGGER.debug(detail)
+
+
+def attach_ns(ns_id, controller_name, cont_id):
+    """
+    attach the namespace_id to specified controller
+
+    :param ns_id: namespace ID
+    :param controller_name: controller name
+    :param cont_id: controller_ID
+    """
+    cmd = f"nvme attach-ns {controller_name} --namespace-id={ns_id} -controllers={cont_id}"
+    try:
+        process.run(cmd, shell=True, ignore_status=True)
+        LOGGER.debug("namespace attached successfully")
+    except process.CmdError as detail:
+        LOGGER.debug("Failed to attach namespace, error=%s", detail)
+    ns_rescan(controller_name)
+
+
+def create_full_capacity_ns(controller_name):
+    """
+    Creates one namespace with full capacity
+
+    :param controller_name: name of the controller like nvme0/nvme1 etc..
+    """
+    if get_current_ns_list(controller_name):
+        raise NvmeException("ns already exist, delete it before creating ")
+    create_one_ns("1", controller_name, get_total_capacity(controller_name))
+
+
+def create_one_ns(ns_id, controller_name, ns_size):
+    """
+    creates a single namespaces with given size and controller_id
+
+    :param ns_id: Namespace ID
+    :param controller_name: name of the controller like nvme0/nvme1 etc..
+    :param ns_size: Size of the namespace that is goin to be created
+    """
+    cmd = f"nvme create-ns {controller_name} --nsze={ns_size} --ncap={ns_size} --flbas=0 -dps=0"
+    try:
+        process.run(cmd, shell=True, ignore_status=True)
+        LOGGER.debug("namespace attached successfully")
+    except process.CmdError as detail:
+        LOGGER.debug("Failed to create namespace, error=%s", detail)
```

Comment:
should not we raise/exit/return if name space fail

Reply:
ok, will re-check on this and change accordingly